PyDSS/ProfileManager/hooks/HDF5.py

```diff
@@ -1,59 +1,54 @@
 from PyDSS.exceptions import InvalidParameter
-from PyDSS.pyLogger import getLoggerTag
-from PyDSS.utils.utils import load_data
+from PyDSS.common import DATE_FORMAT
 from datetime import datetime
 import pandas as pd
 import numpy as np
 import datetime
-import logging
-import toml
 import h5py
+import copy
 import os
 
+class ProfileManager(BaseProfileManager):
 
-
-class ProfileManager:
-
-    def __init__(self,  dssObjects, dssSolver, options, mode="r+"):
-        if options["Logging"]["Pre-configured logging"]:
-            logger_tag = __name__
-        else:
-            logger_tag = getLoggerTag(options)
-        self._logger = logging.getLogger(logger_tag)
-        self.dssSolver = dssSolver
-        self.Objects = dssObjects
-
-        self.profileMapping = load_data(options['Profiles']["Profile mapping"])
-
-        filePath = options['Profiles']["Profile store path"]
-        if os.path.exists(filePath):
-            self._logger.info("Loading existing h5 store")
-            self.store = h5py.File(filePath, mode)
+    def __init__(self,  sim_instance, solver, options, logger, **kwargs):
+        super(ProfileManager, self).__init__(sim_instance, solver, options, logger, **kwargs)
+        self.Objects = kwargs["objects"]
+        if os.path.exists(self.basepath):
+            self.logger.info("Loading existing h5 store")
+            self.store = h5py.File(self.basepath, "r+")
         else:
-            self._logger.info("Creating new h5 store")
-            self.store = h5py.File(filePath, "w")
+            self.logger.info("Creating new h5 store")
+            self.store = h5py.File(self.basepath, "w")
             for profileGroup in PROFILE_TYPES.names():
                 self.store.create_group(profileGroup)
+        self.setup_profiles()
         return
 
     def setup_profiles(self):
         self.Profiles = {}
-        for group, profileMap in self.profileMapping.items():
+        for group, profileMap in self.mapping.items():
             if group in self.store:
                 grp = self.store[group]
                 for profileName, mappingDict in profileMap.items():
                     if profileName in grp:
-                        objects = {x['object'] : self.Objects[x['object']] for x in mappingDict}
-                        self.Profiles[f"{group}/{profileName}"] = TSP(grp[profileName], objects, self.dssSolver,
-                                                                      mappingDict)
+                        objects = {x['object']: self.Objects[x['object']] for x in mappingDict}
+                        self.Profiles[f"{group}/{profileName}"] = Profile(
+                            self.sim_instance,
+                            grp[profileName],
+                            objects,
+                            self.solver,
+                            mappingDict,
+                            self.logger,
+                            **self.kwargs
```

Comment:
I don't see a `self.kwargs` defined. Is this a bug?

Reply:
This depends on the extension. Implemented to provide flexibility for future extensions. It is working as intended